### ISSUE TITLE
bug: Incorrect argument types when calling a compass-evm logic call function

### ISSUE THREAD

**Issue by Vizualni** [palomachain/paloma#476]
# What is happening?
<details>
  <summary>Section description</summary>
<i>Provide as much context as you can. Provide relevant software versions, screenshots, copy & paste error messages and so on.
Give as much context as you can to make it easier for the developers to figure what is happening.</i>
</details>

Saw this log being dropped on telegram:

```
time="2022-09-14T13:44:08Z" level=info msg="relaying messages" messages-to-relay="[450]" queue-name=EVM/eth-main/evm-turnstone-message
time="2022-09-14T13:44:09Z" level=error msg="processing error" chain-reference-id=eth-main error="original error: abi: cannot use array as type struct as argument\nenriched fields:\nmessage id[uint64] = 0x1c2\nmessage type[*types.Message_SubmitLogicCall] = &types.Message_SubmitLogicCall{SubmitLogicCall:(*types.SubmitLogicCall)(0xc000bac640)}\n" message-type="*types.Message_SubmitLogicCall" msg-id=450 queue-name=EVM/eth-main/evm-turnstone-message
```

Pigeon is having an issue calling a logic on the compass-evm. It's expecting the arguments to be a struct, while slice was provided.

# How to reproduce?
<details>
  <summary>Section description</summary>
  <i>Please write detailed steps of what you were doing for this bug to appear.</i>
</details>

Simply put a submit logic call into an evm message queue for execution.

# What is the expected behaviour?
<details>
  <summary>Section description</summary>
<i>If you know, please write down what is the expected behaviour. If you don't know, that's ok. We can have a discussion in comments.</i>
</details>

When adding an logic call to the queue, pigeon must execute it without it returning an error about incorrect argument types.

**Comment by taariq** [comment]
Fixed.
